src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -160,6 +160,9 @@ export class HaConfigDevicePage extends LitElement {
     const batteryState = batteryEntity
       ? this.hass.states[batteryEntity.entity_id]
       : undefined;
+    const batteryIsBinary =
+      batteryState &&
+      (batteryState.state === "off" || batteryState.state === "on");
```

Comment:
You should check the domain of the entity, instead of the states.
<https://github.com/home-assistant/frontend/blob/dev/src/common/entity/compute_state_domain.ts#L4>